classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -928,7 +929,7 @@ protected function claim_actions( $claim_id, $limit, \DateTime $before_date = nu
 		 *
 		 * @param string $order_by_sql
 		 */
-		$order    = apply_filters( 'action_scheduler_claim_actions_order_by', 'ORDER BY attempts ASC, scheduled_date_gmt ASC, action_id ASC' );
+		$order    = apply_filters( 'action_scheduler_claim_actions_order_by', 'ORDER BY priority ASC, attempts ASC, scheduled_date_gmt ASC, action_id ASC' );
```

Comment:
not sure about this query, this query already causes bunch of issues for high traffic sites, adding another order clause will probably worsen it. any ideas around this?

Reply:
perhaps we can use tools like `mysqlslap` to simulate deadlock conditions and see what can we do to improve these

Reply:
Looking at this (benchmarking, also alternatives).

Though I think it's worth noting that while this introduces additional overhead of its own, its potential value only becomes apparent in those situations where the system is under pressure (because when things are running smoothly, priorities will not be needed at all).